changelog.d/764.feature.md

```diff
@@ -0,0 +1,4 @@
+Added experimental `dns_lookup` function. It should be used with caution, since it involves network
```

Comment:
Perhaps we should put it behind a feature flag, at least until we introduce some function result caching mechanism.

Reply:
That makes sense, although doesn't `reverse_dns` also include network calls? Should that one be put behind a network flag too, or does that one utilize some kind of caching?

Reply:
Sorry, just checking if I understood correctly. Based on discussion from: https://github.com/vectordotdev/vrl/issues/720 I thought this function was supposed to be undocumented (at least until caching is sorted out), similar to `reverse_dns`.

I can add a warning to make it more visible that it involves network calls and should be avoided in most cases if we want to have it documented in the end though.

Reply:
for the above question cc @jszwedko

Reply:
👍 I'm still fine with adding this but not documenting it, for the time being, until we can sort out a caching strategy.